cfme/common/provider_views.py

```diff
@@ -248,5 +286,146 @@ def is_displayed(self):
 
     toolbar = View.nested(ProviderToolBar)
     sidebar = View.nested(ProviderSideBar)
-    entities = View.nested(ProviderEntities)
+    items = View.nested(ProviderItems)
     paginator = View.nested(PaginationPane)
+
+
+class BeforeFillMixin(object):
```

Comment:
without having reviewed the rest yet I think I'm going to like this implementation for before_fill.